app/controlplane/internal/service/workflow.go

```diff
@@ -73,12 +75,25 @@ func (s *WorkflowService) Update(ctx context.Context, req *pb.WorkflowServiceUpd
 		return nil, handleUseCaseErr(err, s.log)
 	}
 
+	var contractID *string
+	if req.ContractName != nil {
+		c, err := s.contractUC.FindByNameInOrg(ctx, currentOrg.ID, *req.ContractName)
+		if err != nil {
+			return nil, handleUseCaseErr(err, s.log)
+		} else if c == nil {
+			return nil, biz.NewErrNotFound("contract")
+		}
+
+		cid := c.ID.String()
+		contractID = &cid
+	}
+
 	updateOpts := &biz.WorkflowUpdateOpts{
 		Project:     req.Project,
 		Team:        req.Team,
 		Public:      req.Public,
 		Description: req.Description,
-		ContractID:  req.SchemaId,
+		ContractID:  contractID,
```

Comment:
I decided not to change anything in the biz/data layer. basically biz/data still uses IDs internally and it's the server layer the one that does the translation if needed.

Otherwise we get into the problematic of wether we should change all IDs, including organizationIDs to names in the business layer and it will be a huge rewrite, and tbh I am not sure of the performance implications either (due to the current indexes)